tests/console/snapper_cleanup_timeline.pm

```diff
@@ -0,0 +1,61 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Configure snapper and verify that timeline cleanup algorithm behaves accordingly.
+# Maintainer: QA SLE yast team <qa-sle-yast@suse.de>
+
+use strict;
+use warnings;
+use base 'y2_installbase';
+use testapi;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+sub pre_run_hook {
+    my $test_data = get_test_suite_data();
+    select_console 'root-console';
+    record_info("Configuration", "Configure snapper.");
+    foreach my $param (keys %{$test_data->{snapper_config}}) {
+        assert_script_run("snapper set-config $param=$test_data->{snapper_config}->{$param}",
+            fail_message => "Snapper configuration failed for parameter $param");
+    }
+}
+
+sub run {
+    record_info("Check quota", "Verify that the percentage of root filesystem quota is less than 50%");
+    my $convert_numeric = qr/[^0-9.]+/;
+    my $disk_space      = script_output("df -h /  | awk \'NR==2 {print \$4}\'");
+    $disk_space =~ s/$convert_numeric//;
+    my $qgroup_space = script_output("btrfs qgroup show -F --sync --si / | awk \'NR==3 {print \$2}\'");
+    $qgroup_space =~ s/$convert_numeric//;
+    die "Root filesystem quota take more than 50% of root disk space" if ($qgroup_space >= 0.5 * $disk_space);
+
+    record_info("Space > 20%", "Ensure that free space is more than 20%. 
+	    Create a snapshot with timeline cleanup algorithm and make sure that it is not erased by the algorithm");
+    $disk_space = script_output("df -h /  | awk \'NR==2 {print \$5}\'");
+    die "Used disk space is more than 80%" if ($disk_space >= 80);
+    assert_script_run("snapper create --description \"timeline\" --cleanup-algorithm timeline",
+        fail_message => "Timeline snapshot failed to be created");
+    assert_script_run("snapper cleanup timeline",
+        fail_message => "Timeline cleanup algorithm failed to run");
+    assert_script_run("snapper ls | grep timeline",
+        fail_message => "No timeline snapshot found");
+
+    record_info("Space < 20%", "Fill up disk space and make sure that the created snapshot gets erased by 
+	    timeline cleanup algorithm");
+    assert_script_run("dd if=/dev/urandom of=/tmp/blob bs=1M count=30500", timeout => 900,
```

Comment:
Can you consider to use the default for the block size here? i cant find a good reason to use 1M which if i am right it will slow down the operation. dd is already slow. What do you think?

Reply:
Performance depends on the underlying block size of the partition, default value of the `dd` command is 512 bytes, I think is 16KB for btrfs, but I'm not sure, we can use `stat -fc %s .` to check, but still a minor thing I guess.

Reply:
I changed bs=10M but the command is insignificantly faster.